quarkchain/evm/messages.py

```diff
@@ -351,7 +361,31 @@ def apply_transaction(state, tx: transactions.Transaction, tx_wrapper_hash):
         state.get_balance(tx.sender, token_id=tx.gas_token_id)
         >= tx.startgas * tx.gasprice
     )
-    state.delta_token_balance(tx.sender, tx.gas_token_id, -tx.startgas * tx.gasprice)
+    original_gas_token_id = tx.gas_token_id
+    refund_rate = 100
+    if tx.gas_token_id != state.genesis_token_id:
+        refund_rate, gasprice = state.get_gas_utility_info(
```

Comment:
`gasprice` can be easily confused with `tx.gasprice`. how about naming it as `converted_genesis_token_gas_price`?